core.md

```diff
@@ -235,6 +244,156 @@ default can save a significant amount of memory, particularly if you
 have a large number of lines embedded in an array. It also
 improves cache behavior since the data is contiguous in memory.
 
+## Alignment and Padding
+
+The alignment of a struct type member field is determined by its type.
+
+### Alignment: Primitive Types
+
+For primitive types, the alignment equals the byte length of the type.
+For the `any`, `string`, and `object` types, the byte length is
+implementation-dependent. (However, that doesn't matter much as it's not
+content-observable. See the [section on opacity](#opacity) for details.)
+
+The byte length and thus alignment of the other types is:
+
+Type    | Byte Length / Alignment
+--------|------------
+uint8   | 1
+int8    | 1
+float32 | 4
+uint16  | 2
+int16   | 2
+uint32  | 4
+int32   | 4
+float64 | 8
+
+### Alignment: Nested Structs
+
+For embedded structs, the alignment is determined by the largest contained
+type. If the embedded struct only contains fields with primitive types, its
+alignment is that of the field with the largest primitive type. If it itself
+contains embedded structs, its alignment is that of the largest embedded
+struct.
+
+### Padding
+
+To ensure the alignment rules described above hold, padding is inserted between
+fields of a struct and struct array elements. There are two ways in which this
```

Comment:
You could be more specific here, not that it matters. i.e., to align each field up to its natural alignment.